computer_vision/mosaic_augmentation.py

```diff
@@ -0,0 +1,180 @@
+        file_root = f"{OUTPUT_DIR}/{file_name}_MOSAIC_{letter_code}"
+        cv2.imwrite(f"{file_root}.jpg", new_image,
+                    [cv2.IMWRITE_JPEG_QUALITY, 85])
+        print(f'Successed {index+1}/{NUMBER_IMAGES} with {file_name}')
+        annos_list = []
+        for anno in new_annos:
+            width = anno[3] - anno[1]
+            height = anno[4] - anno[2]
+            x_center = anno[1] + width/2
+            y_center = anno[2] + height/2
+            obj = f'{anno[0]} {x_center} {y_center} {width} {height}'
+            annos_list.append(obj)
+        with open(f"{file_root}.txt", "w") as outfile:
+            outfile.write("\n".join(line for line in annos_list))
+
+
+def get_dataset(label_dir: str, img_dir: str) -> list:
+    """
+    - label_dir <type: str>: Path to label include annotation of images
+    - img_dir <type: str>: Path to folder contain images
+    Return <type: list>: List of images path and labels
+    >>> pass  # A doctest is not possible for this function.
+    """
+    img_paths = []
+    labels = []
+    for label_file in glob.glob(os.path.join(label_dir, '*.txt')):
+        label_name = label_file.split('/')[-1].rsplit('.', 1)[0]
+        with open(label_file, 'r') as in_file:
+            obj_lists = in_file.readlines()
+        img_path = os.path.join(img_dir, f'{label_name}.jpg')
+
+        boxes = []
+        for obj_list in obj_lists:
+            obj = obj_list.rstrip('\n').split(' ')
+            xmin = float(obj[1]) - float(obj[3])/2
+            ymin = float(obj[2]) - float(obj[4])/2
+            xmax = float(obj[1]) + float(obj[3])/2
+            ymax = float(obj[2]) + float(obj[4])/2
+
+            boxes.append([int(obj[0]), xmin, ymin, xmax, ymax])
+        if not boxes:
+            continue
+        img_paths.append(img_path)
+        labels.append(boxes)
+    return img_paths, labels
+
+
+def update_image_and_anno(all_img_list: list, all_annos: list, idxs: int, output_size: int, scale_range: int, filter_scale: int = 0.) -> list:
+    """
+    - all_img_list <type: list>: list of all images
+    - all_annos <type: list>: list of all annotations of specific image
+    - idxs <type: list>: index of image in list
+    - output_size <type: tuple>: size of output image (Height, Width)
+    - scale_range <type: tuple>: range of scale image
+    - filter_scale <type: float>: the condition of downscale image and bounding box
+    Return:
+        - output_img <type: narray>: image after resize
+        - new_anno <type: list>: list of new annotation after scale
+        - path[0] <type: string>: get the name of image file
+    >>> pass  # A doctest is not possible for this function.
+    """
+    output_img = np.zeros([output_size[0], output_size[1], 3], dtype=np.uint8)
+    scale_x = scale_range[0] + \
+        random.random() * (scale_range[1] - scale_range[0])
+    scale_y = scale_range[0] + \
+        random.random() * (scale_range[1] - scale_range[0])
```

Comment:
PEP8 says we should avoid backslash line continuation in Python because whitespace to the right of the backslash breaks the script in a change that is invisible to the reader.  I believe that psf/black will reformat as...
```suggestion
    scale_x = scale_range[0] + random.random() * (
        scale_range[1] - scale_range[0]
    )
    scale_y = scale_range[0] + random.random() * (
        scale_range[1] - scale_range[0]
    )
```